BPRBE/Persistence/MongoDependencyRepository.cs

```diff
@@ -0,0 +1,51 @@
+﻿using BPRBE.Config;
+using BPRBE.Constants;
+using BPRBE.Models.Persistence;
+using Microsoft.Extensions.Options;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace BPRBE.Persistence;
+
+public class MongoDependencyRepository : IMongoDependencyRepository
+{
+    private readonly IMongoCollection<ArchitecturalModel> _dependenciesRuleCollection;
+    private readonly IMongoCollection<BsonDocument> _sequenceCollection;
+
+    public MongoDependencyRepository(IOptions<DatabaseConfig> databaseSettings)
+    {
+        var mongoClient = new MongoClient(databaseSettings.Value.ConnectionString);
+        var mongoDatabase = mongoClient.GetDatabase(databaseSettings.Value.DatabaseName);
+
+        _dependenciesRuleCollection =
+            mongoDatabase.GetCollection<ArchitecturalModel>(databaseSettings.Value.DependenciesRuleCollectionName);
+        _sequenceCollection = mongoDatabase.GetCollection<BsonDocument>(databaseSettings.Value.DependenciesRuleCollectionName);
+    }
+
+    public async Task<IList<ArchitecturalModel>> GetArchitecturalModelsAsync()
+    {
+        return await (await _dependenciesRuleCollection.FindAsync(_ => true)).ToListAsync();
+    }
+    public async Task<ArchitecturalModel?> GetArchitecturalModelByName(ArchitecturalModel model)
+    {
+        return await (await _dependenciesRuleCollection.FindAsync(x => x.Name.Equals(model.Name))).FirstOrDefaultAsync();
+    }
```

Comment:
```suggestion
    public async Task<IList<ArchitecturalModel>> GetArchitecturalModelsAsync()
    {
        return await _dependenciesRuleCollection
            .Find(_ => true)
            .ToListAsync();
    }
    
    public async Task<ArchitecturalModel?> GetArchitecturalModelByName(ArchitecturalModel model)
    {
        return await _dependenciesRuleCollection
            .Find(x => x.Name == model.Name)
            .FirstOrDefaultAsync();
    }
```

I would also personally not return the interface IList here but rather the actual implementation, although I did not suggest that. Up to you.
Could also cosider making variables for the filters like this:

        var filter = Builders<ArchitecturalModel>.Filter.Empty;

        return await _dependenciesRuleCollection
            .Find(filter)
            .ToListAsync();